agent/eni/netwrapper/netwrapper_windows_test.go

```diff
@@ -0,0 +1,155 @@
+//go:build windows
+// +build windows
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package netwrapper
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	dummyIfAlias              = "Ethernet 2"
+	dummyIfIndex              = 9
+	dummyMAC                  = "06-34-BB-C8-64-C5"
+	networkAdapterValidResult = `[
+  {
+    "ifAlias": "%s",
+    "ifIndex": %d,
+    "MacAddress": "%s",
+    "ActiveMaximumTransmissionUnit": 1500
+  }
+]`
+	networkAdapterExpectedValidResult []net.Interface
+	networkAdapterInvalidResult       = ""
+)
+
+func init() {
+	networkAdapterValidResult = fmt.Sprintf(networkAdapterValidResult, dummyIfAlias, dummyIfIndex, dummyMAC)
+
+	parsedMAC, _ := net.ParseMAC(dummyMAC)
+	networkAdapterExpectedValidResult = []net.Interface{
+		{
+			Index:        dummyIfIndex,
+			MTU:          1500,
+			Name:         dummyIfAlias,
+			HardwareAddr: parsedMAC,
+		},
+	}
+
+}
+
+// Supporting methods for testing getAllNetworkInterfaces method.
+// fakeExecCommandForNetworkAdaptersSuccess returns a valid output from exec cmd.
+func fakeExecCommandForNetworkAdaptersSuccess(command string, args ...string) *exec.Cmd {
+	cs := []string{"-test.run=TestNetworkAdaptersProcessSuccess", "--", command}
+	cs = append(cs, args...)
+	cmd := exec.Command(os.Args[0], cs...)
+	cmd.Env = []string{"GO_WANT_HELPER_PROCESS=1"}
+	return cmd
+}
+
+// TestNetworkAdaptersProcessSuccess is invoked from fakeExecfakeExecCommandForNetworkAdaptersSuccessCommand
+// to return the network adapter information.
+func TestNetworkAdaptersProcessSuccess(t *testing.T) {
+	if os.Getenv("GO_WANT_HELPER_PROCESS") != "1" {
+		return
+	}
+	fmt.Fprintf(os.Stdout, networkAdapterValidResult)
+	os.Exit(0)
+}
+
+// fakeExecCommandForNetworkAdaptersInvalidOutput returns an invalid output from exec cmd.
+func fakeExecCommandForNetworkAdaptersInvalidOutput(command string, args ...string) *exec.Cmd {
```

Comment:
nit -- can we call this `mock` instead of `fake`?  (That goes for other instances of the same.)